### PR TITLE
Scan: Validator and proposal message restructure

### PR DESCRIPTION
**Pull request by Pitchanai**
Fixed: #2699 
Fixed: #2700 

## Implementation details
- Restructure validator message
- Restructure proposal message

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by pzshine — APPROVED** [review]

**Comment by pzshine** [comment]
Could you wire up the data(moniker, proposal name, etc)?

File: scan/src/components/msgs/Msg.re [hunk 1]
```diff
@@ -237,96 +237,11 @@ let make = (~msg: TxSub.Msg.t, ~width: int) => {
...
-                  color=Colors.gray7
-                  weight=Text.Regular
-                  code=true
-                  nowrap=true
-                  block=true
-                  ellipsis=true
-                />}
-         </div>
-       </div>
+     | AddReporterMsgSuccess({validator, reporter}) => <ValidatorMsg.AddReporter reporter />
```
Comment (evilpeach) [inline scan/src/components/msgs/Msg.re]: remove unused validator

File: scan/src/components/msgs/ProposalMsg.re [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  [@react.component]
+  let make = (~proposer, ~title) => {
+    <div
+      className={Css.merge([
+        CssHelper.flexBox(~wrap=`nowrap, ()),
+        CssHelper.overflowHidden,
+        Styles.msgContainer,
+      ])}>
+       <Text value=title size=Text.Md nowrap=true block=true /> </div>;
+      // TODO: Proposal ID
```
Comment (evilpeach) [inline scan/src/components/msgs/ProposalMsg.re]: Let's implement it.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/subscriptions/TxSub.re [hunk 3]
```diff
@@ -869,6 +869,13 @@ module Msg = {
         delegatorAddress: json |> at(["msg", "delegator_address"], string) |> Address.fromBech32,
       };
     };
+
+    let decodeFail = json => {
+      JsonUtils.Decode.{
+        validatorAddress: json |> at(["msg", "validator_address"], string) |> Address.fromBech32,
+        delegatorAddress: json |> at(["msg", "delegator_address"], string) |> Address.fromBech32,
+      };
+    };
```
Comment (evilpeach) [inline scan/src/subscriptions/TxSub.re]: It's duplicated.

File: scan/src/subscriptions/TxSub.re [hunk 4]
```diff
@@ -890,13 +897,33 @@ module Msg = {
...
+        proposer: json |> at(["msg", "proposer"], string) |> Address.fromBech32,
+        title: json |> at(["msg", "content", "title"], string),
+        description: json |> at(["msg", "content", "description"], string),
+        initialDeposit: json |> at(["msg", "initial_deposit"], list(Coin.decodeCoin)),
+        proposalID: json |> at(["extra", "proposal_id"], ID.Proposal.fromJson),
+      };
+    };
+
+    let decodeFail = json => {
+      Js.log(json);
```
Comment (evilpeach) [inline scan/src/subscriptions/TxSub.re]: remove log

File: scan/src/subscriptions/TxSub.re [hunk 5]
```diff
@@ -1095,11 +1162,11 @@ module Msg = {
...
     | SetWithdrawAddressMsgFail(set) => set.delegatorAddress
-    | SubmitProposalMsgSuccess(proposal)
+    | SubmitProposalMsgSuccess(proposal) => proposal.proposer
     | SubmitProposalMsgFail(proposal) => proposal.proposer
-    | DepositMsgSuccess(deposit)
+    | DepositMsgSuccess(deposit) => deposit.depositor
     | DepositMsgFail(deposit) => deposit.depositor
-    | VoteMsgSuccess(vote)
+    | VoteMsgSuccess(vote) => vote.voterAddress
     | VoteMsgFail(vote) => vote.voterAddress
```
Comment (evilpeach) [inline scan/src/subscriptions/TxSub.re:1173]: Why do you change it?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/subscriptions/TxSub.re (hunk 5 above)
Comment (evilpeach) [reply scan/src/subscriptions/TxSub.re:1173]: @Pitchanai

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Self Delegation" size=Heading.H5 marginBottom=8 />
+        <AmountRender coins=[validator.selfDelegation] pos=AmountRender.TxIndex />
+      </Col.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
+        <Heading value="Details" size=Heading.H5 marginBottom=8 />
+        <Text size=Text.Lg weight=Text.Thin value={validator.details} />
+      </Col.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: The `mb` should be removed, there is extra space on the bottom of box.

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+        <Heading value="Min Self Delegation" size=Heading.H5 marginBottom=8 />
+        {switch (validator.minSelfDelegation) {
+         | Some(minSelfDelegation') =>
+           <AmountRender coins=[minSelfDelegation'] pos=AmountRender.TxIndex />
+         | None => <Text value="Unchanged" code=true size=Text.Lg weight=Text.Thin />
+         }}
+      </Col.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: The `mb` should be removed, there is extra space on the bottom of box.

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module UnjailMsg = {
+  [@react.component]
+  let make = (~unjail: TxSub.Msg.Unjail.success_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please remove `mb`

File: scan/src/components/tx-index/IndexProposalMsg.re [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    <Row.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Proposer" size=Heading.H5 marginBottom=8 />
+        <AddressRender position=AddressRender.Subtitle address={proposal.proposer} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Proposal ID" size=Heading.H5 marginBottom=8 />
+        <TypeID.Proposal position=TypeID.Subtitle id={proposal.proposalID} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexProposalMsg.re]: Please replace `mb` with `mbSm` to apply the marginbottom only on mobile screen

File: scan/src/components/tx-index/IndexProposalMsg.re [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        <Heading value="Title" size=Heading.H5 marginBottom=8 />
+        <Text size=Text.Lg value={proposal.title} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexProposalMsg.re]: Please remove `mb`

File: scan/src/components/tx-index/IndexProposalMsg.re [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+module SubmitProposalFailMsg = {
+  [@react.component]
+  let make = (~proposal: TxSub.Msg.SubmitProposal.fail_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
+        <Heading value="Proposer" size=Heading.H5 marginBottom=8 />
+        <AddressRender position=AddressRender.Subtitle address={proposal.proposer} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexProposalMsg.re]: Please replace `mb` with `mbSm` to apply the marginbottom only on mobile screen

File: scan/src/components/tx-index/IndexProposalMsg.re [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+        <Heading value="Title" size=Heading.H5 marginBottom=8 />
+        <Text size=Text.Lg value={proposal.title} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexProposalMsg.re]: Please remove `mb`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Min Self Delegation" size=Heading.H5 marginBottom=8 />
+        <AmountRender coins=[validator.minSelfDelegation] pos=AmountRender.TxIndex />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Self Delegation" size=Heading.H5 marginBottom=8 />
+        <AmountRender coins=[validator.selfDelegation] pos=AmountRender.TxIndex />
+      </Col.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: The default `col` is `Col.Twelve`, so we can use `<Col.Grid>`

File: scan/src/components/tx-index/IndexValidatorMsg.re (hunk 6 above)
Comment (pzshine) [reply scan/src/components/tx-index/IndexValidatorMsg.re]: The default `col` is `Col.Twelve`, so we can use `<Col.Grid>`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Validator Address" size=Heading.H5 marginBottom=8 />
+        <AddressRender
+          position=AddressRender.Subtitle
+          address={validator.sender}
+          accountType=`validator
+        />
+      </Col.Grid>
+      <Col.Grid col=Col.Twelve mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please remove `mb`, and `col=Col.Twelve`

File: scan/src/components/tx-index/IndexValidatorMsg.re (hunk 7 above)
Comment (pzshine) [reply scan/src/components/tx-index/IndexValidatorMsg.re]: The default `col` is `Col.Twelve`, so we can use `<Col.Grid>`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module AddReporterMsg = {
+  [@react.component]
+  let make = (~address: TxSub.Msg.AddReporter.success_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please replace `mb` with `mbSm`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+        <Heading value="Validator" size=Heading.H5 marginBottom=8 />
+        <AddressRender
+          position=AddressRender.Subtitle
+          address={address.validator}
+          accountType=`validator
+        />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re:196]: Please remove `mb`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module AddReporterFailMsg = {
+  [@react.component]
+  let make = (~address: TxSub.Msg.AddReporter.fail_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please add `mbSm`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module RemoveReporterMsg = {
+  [@react.component]
+  let make = (~address: TxSub.Msg.RemoveReporter.success_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please add `mbSm`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module RemoveReporterFailMsg = {
+  [@react.component]
+  let make = (~address: TxSub.Msg.RemoveReporter.fail_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please add `mbSm`

File: scan/src/components/tx-index/IndexValidatorMsg.re [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module SetWithdrawAddressMsg = {
+  [@react.component]
+  let make = (~set: TxSub.Msg.SetWithdrawAddress.t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six>
```
Comment (pzshine) [inline scan/src/components/tx-index/IndexValidatorMsg.re]: Please add `mbSm`

**Review by pzshine — CHANGES_REQUESTED** [review]

**Review by pzshine — APPROVED** [review]

**Pull request merged**